table_one.py

```
@@ -369,15 +384,40 @@ def generate_table(df, selected_vars, group_col, var_meta, or_style='all_levels'
+        # Determine columns to subset for hashing
+        cols_to_hash = vars_list + ([group_col] if group_col and group_col != "None" else [])
+        
+        data_subset = df[cols_to_hash]
+        data_hash = hash(np.sum(pd.util.hash_pandas_object(data_subset).values))
+        cache_key = f"table1_{data_hash}_{hash(tuple(vars_list))}_{group_col}_{or_style}"
+        
+        cached_result = COMPUTATION_CACHE.get(cache_key)
+        if cached_result:
+            logger.info("Cache hit for Table One generation")
+            return cached_result
+    except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)